src/geom/Transform.cc

```diff
@@ -118,6 +120,25 @@ Transform<ToEndpoint, FromEndpoint> Transform<FromEndpoint, ToEndpoint>::getInve
 }
 
 template <typename FromEndpoint, typename ToEndpoint>
+Eigen::MatrixXd Transform<FromEndpoint, ToEndpoint>::getJacobian(FromPoint const &x) const {
+    try {
+        size_t const nIn = _fromEndpoint.getNAxes();
+        size_t const nOut = _toEndpoint.getNAxes();
+        std::vector<double> const point = _fromEndpoint.dataFromPoint(x);
+
+        Eigen::MatrixXd jacobian(nOut, nIn);
+        for (size_t i = 0; i < nOut; ++i) {
+            for (size_t j = 0; j < nIn; ++j) {
+                jacobian(i, j) = _frameSet->rate(point, i + 1, j + 1);
+            }
+        }
+        return jacobian;
+    } catch (std::bad_alloc const &e) {
+        std::throw_with_nested(pex::exceptions::MemoryError("Could not allocate Jacobian."));
```

Comment:
This is mainly for API consistency. We have a dedicated `MemoryError` exception, so presumably we should emit it in preference to `std::bad_alloc`. Although perhaps now that pybind11 gives us better exception wrapping than Swig did, we should push for deprecation of `MemoryError` as part of DM-9435.

As for constructing the exception, it looks like `LSST_EXCEPT` evaluates to an exception object rather than a throw statement, so I can do that.